### ISSUE TITLE
how to optimize the parameters of model and loss at the same time? 

### ISSUE THREAD

**Issue by weihua04** [pytorch/pytorch#18153]
I use the code below to optimize the parameters both in the model and loss.
     "trainable = [{'params': loss.parameters(), 'lr': args.lr},
      {'params': model.parameters(), 'lr': args.lr}] "
and I found a strange phenomenon , if i use the above optimizing method , for training several epoches, the loss descend from 11.7 to 10.8 and just stay there .
but if I put all the parameters in the model ,and use the following code . the loss descend from 11.7 to 1.3
    "trainable = [{'params': model.parameters(), 'lr': args.lr}] "

ps: the weights in the loss using the normalize, because if i didn't use normalize ,the loss sometimes become "nan"

**Comment by soumith** [comment]
if you can give a fuller reproduction, we can take a look. this does not sound right.

**Comment by weihua04** [comment]
sorry,my mistake ,i use the cosface loss and didn't introduce right parameter of cosface,
thank u all the same for reply

**Comment by weihua04** [comment]
> if you can give a fuller reproduction, we can take a look. this does not sound right.

sorry,my mistake ,i use the cosface loss and didn't introduce right parameter of cosface,
thank u all the same for reply

**Comment by pukkinming** [comment]
Normalizing before applying cosface loss helps too for my situation.
